diff.go

```diff
@@ -0,0 +1,193 @@
+			return nil, nil, nil, -1, fmt.Errorf("Error in creating leftunique6s: %w", err)
+		}
+	}
+	leftunique = append(leftunique4s, leftunique6s...)
+
+	// Create rightunique
+	var rightunique4s []*net.IPNet
+	if len(right4s) > 0 {
+		rightunique4s, err = remove4(copy4s(right4s), copy4s(left4s))
+		if err != nil {
+			return nil, nil, nil, -1, fmt.Errorf("Error in creating rightunique4s: %w", err)
+		}
+	}
+	var rightunique6s []*net.IPNet
+	if len(right6s) > 0 {
+		rightunique6s, err = remove6(copy6s(right6s), copy6s(left6s))
+		if err != nil {
+			return nil, nil, nil, -1, fmt.Errorf("Error in creating rightunique6s: %w", err)
+		}
+	}
+	rightunique = append(rightunique4s, rightunique6s...)
+
+	// Create inboth
+	var inboth4s []*net.IPNet
+	if len(left4s) > 0 {
+		inboth4s, err = subset4(left4s, right4s)
+		if err != nil {
+			return nil, nil, nil, -1, fmt.Errorf("Error in creating inboth4s: %w", err)
+		}
+	}
+	var inboth6s []*net.IPNet
+	if len(left6s) > 0 {
+		inboth6s, err = subset6(left6s, right6s)
+		if err != nil {
+			return nil, nil, nil, -1, fmt.Errorf("Error in creating inboth6s: %w", err)
+		}
+	}
+	inboth = append(inboth4s, inboth6s...)
+
+	if len(inboth) == 0 {
+		status = 4
+	} else if len(leftunique) == 0 && len(rightunique) == 0 {
+		status = 0
+	} else if len(leftunique) == 0 {
+		status = 3
+	} else if len(rightunique) == 0 {
+		status = 2
+	} else {
+		status = 1
+	}
+	return leftunique, inboth, rightunique, status, nil
+}
+
+// DiffCIDRs accepts two lists of mixed CIDR blocks and return three lists of
+// CIDRs that  only exists in first list, second lost and in both lists, together with a status number.
+func DiffCIDRs(leftcidrs, rightcidrs []string) (leftreturn []string, bothreturn []string, rightreturn []string, status int, err error) {
+	if leftcidrs == nil || rightcidrs == nil {
+		return nil, nil, nil, -2, fmt.Errorf("Missing input data in DiffCIDRs()")
+	}
+	if len(leftcidrs) == 0 && len(rightcidrs) == 0 {
+		return make([]string, 0), make([]string, 0), make([]string, 0), 5, nil
+	}
+	if len(leftcidrs) == 0 {
+		return make([]string, 0), make([]string, 0), rightcidrs, 3, nil
+	}
+	if len(rightcidrs) == 0 {
+		return leftcidrs, make([]string, 0), make([]string, 0), 2, nil
+	}
+
+	var leftnetworks []*net.IPNet
+	for _, cidr := range leftcidrs {
+		_, network, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, nil, nil, -1, err
+		}
+		leftnetworks = append(leftnetworks, network)
+	}
+	var rightnetworks []*net.IPNet
+	for _, cidr := range rightcidrs {
+		_, network, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, nil, nil, -1, err
+		}
+		rightnetworks = append(rightnetworks, network)
+	}
+
+	leftNets, bothNets, rightNets, status,  err := DiffIPNets(leftnetworks, rightnetworks)
+	if err != nil {
+		return nil, nil, nil, status, err
+	}
+
+	// Handle the situation empty data
+	if len(leftNets) == 0 {
+		leftreturn = make([]string, 0)
+	} else {
+		leftreturn = ipNets(leftNets).toCIDRs()
```

Comment:
Shouldn't `toCIDRs()` handle the case of zero-length `leftNets`? Same below for `rightNets` and `bothNets`.
This to avoid all the `if len(..) == 0`-cases.

Reply:
toCIDRs() handle zero-length but the in that case return 'nil' rather than a empty object.